src/admin/components/Ballot.tsx

```diff
@@ -0,0 +1,106 @@
+import classNames from "classnames";
+import { useDispatch } from "react-redux";
+import { useState } from "react";
+
+import { Ballot } from "../../models";
+import { deleteBallot, updateBallot } from "../../network/ballotApi";
+import { BallotForm } from "../sections/BallotForm"
+import { Jwt } from "../../network/jwt";
+import { FetchError } from "../../network/request";
+import styles from "./Ballot.module.scss";
+import { Button } from "./Button";
+
+interface BallotProps {
+  jwt: Jwt;
+  ballot: Ballot;
+  onDelete(): void;
+  onUpdate(): void;
+}
+
+function BallotView({ jwt, ballot, onDelete, onUpdate }: BallotProps) {
+  const [updateFormVisible, setUpdateFormVisible] = useState(false);
+  const dispatch = useDispatch();
+
+  function clickDeleteBallot() {
+    if (confirm("Do you really want to delete this ballot?")) {
+      deleteBallot(jwt, ballot._id)
+        .then(onDelete)
```

Comment:
On delete, the ballots won't update/refetch. For some reason, the `onDelete` doesn't seem to be firing :/